context-free.md

```diff
@@ -2,21 +2,77 @@
+understanding of computers. Specifically, this program seems ideal for artists
+who want to do recursive "math art" like fractals. This appears to be an easy
+alternative to writing things like pure PostScript.
+
+
 
 ## What is easy to do in this language? Why is it easy?
 
+Things like fractals are easy to do in ContextFree. This is because the its
+function-like declarations of rules and paths enable easy reuse and scaling
+of shapes in a programmatic way. Although all art generated in this program can
+be generated by other computer means, this program makes it much more
+accessible to both programmers and artists alike by abstracting out complex
+details related to rendering.
+
+
 
 ## What is hard to do in this language? Why is it hard?
 
+This language appears to be designed to programatically do vector based art,
+similar to SVG files and images generated by software like Adobe Flash and
+Illustrator. While vector art works great for scalable image, (such as posters,
+banners, wallpapers, clipart, fonts, etc.), by defining all shapes and lines in
+terms of math, the user loses control over the individual pixels. Although
+images can be converted to a typical format like a PNG or JPG, this is a result
+of sampling and not individual pixel manipulation as is often done in programs
+like Photoshop. Thus, since everything is very mathematical, it would be
+challenging to design something with a sketch-like or hand-drawn appearance.
+This is because this function is beyond the scope of a DSL built for
+programmatically generated vector based art.
+
+
 
 ## How did you learn how to program in this language?
 
+I used a combination of the sample _welcome.cfdg_ file that appeared when the
+ContextFree opened, the documentation available on the internet, and trial and
+error. The documentation was initially challenging to use since we use an older
+version of the software with slightly different syntax, especially for
+defining paths. Although the old documentation still exists on the web
+[here](http://www.contextfreeart.org/mediawiki/index.php/Version_2_Syntax#Paths)
+it wasn't linked to directly in ContextFree's help menu.
+
+Specifically, I used trial and error to discover things such as the difference
+between _LINETO_ and _LINEREL_, the importance of parameter ordering (e.g.
+SQUARE [r 45 s 1 2] is very different than SQUARE [s 1 2 r 45]), and that Y+ is
+up, unlike monitor coordinates where Y+ is often down.
+
+
+
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+ContextFree interprets a provided script and renders an appropriate image. The
+rendered image can then be modified and re-rendered via the script or exported
+into typical image file format.
```

Comment:
How do you think, or how does it make sense, the program uses the script to render the image? Is there a model of the information written in the DSL that looks like something you have seen before?